src/mono/mono/mini/mini-amd64.c

```diff
@@ -988,7 +988,8 @@ get_call_info (MonoMemPool *mp, MonoMethodSignature *sig)
 				/*
 				 * We need to set this even when sig->pinvoke is FALSE, because the `cinfo` gets copied to the
 				 * `cfg->arch` on the first pass. However, later in `amd64_handle_swift_return_buffer_reg` we 
-				 * condition the Swift return buffer handling only to P/Invoke calls. 
+				 * condition the Swift return buffer handling only to P/Invoke calls. This however can trigger
+				 * a false positive in some scenarios where the Swift return buffer is not needed. 
```

Comment:
What are those false positive triggering scenarios?

Reply:
Mostly cases where struct is large enough that it would normally (outside swiftcall) be returned by a reference but small enough that during swiftcall it is returned via registers.